src/command/subgraph/publish.rs

```diff
@@ -54,8 +58,15 @@ impl Publish {
         client_config: StudioClientConfig,
         git_context: GitContext,
     ) -> RoverResult<RoverOutput> {
+        if self.no_url && self.routing_url.is_some() {
+            return Err(RoverError::new(anyhow!(
+                "You cannot use --no-url and --routing-url at the same time."
+            )));
+        }
+
         // if --allow-invalid-routing-url is not provided, we need to inspect
-        // the URL and possibly prompt the user to publish
+        // the URL and possibly prompt the user to publish. this does nothing
+        // if the routing url is not provided.
         if !self.allow_invalid_routing_url {
```

Comment:
```suggestion
        // the URL and possibly prompt the user to publish. this does nothing
        // if the routing url is not provided.
        if !self.no_url && !self.allow_invalid_routing_url {
```
Why not skip this logic entirely if `--no-url` is passed?

Reply:
I'm fine with that change — it doesn't really matter but I guess it's more complete.

Reply:
should this be || ? I m not sure I follow.